tests/group_functions/type_coverage.h

```diff
@@ -0,0 +1,708 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provide common functions for type coverage
```

Comment:
Should be moved somewhere in `util`?

Reply:
Or actually is this a copy from `tests/common/type_coverage.h`???

Reply:
Like it is said in the beginning of the file it is partial copy with `marray` removed and two additional functions added. Both producing additions but also changes with guards to switch off parts of `common/type_coverage.h` and duplicating its partially here are bad IMO, but I decided that this way is a bit better.